hspec-core/src/Test/Hspec/Core/Extension.hs

```diff
@@ -0,0 +1,71 @@
+module Test.Hspec.Core.Extension (
+-- * Register command-line options
+  Option
+, OptionSetter
+, registerOption
+, flag
+, option
+, argument
+
+-- * Register tree transformations
+, runIO
+, modifyConfig
+, registerTransformation
+
+-- * Types
+-- |
+-- Operations on these types are provided by the following modules
+-- respectively:
+--
+-- * "Test.Hspec.Core.Extension.Config"
+-- * "Test.Hspec.Core.Extension.Item"
+-- * "Test.Hspec.Core.Extension.Spec"
+-- * "Test.Hspec.Core.Extension.Tree"
+, Config
+, Item
+, SpecM
+, SpecWith
+, SpecTree
+) where
+
+import           Prelude ()
+import           Test.Hspec.Core.Compat
+
+import qualified Data.CallStack as CallStack
+
+import qualified GetOpt.Declarative as Declarative
+import           Test.Hspec.Core.Spec (SpecM, SpecWith, runIO)
+import qualified Test.Hspec.Core.Spec as Core
+import qualified Test.Hspec.Core.Config.Definition as Core
+import           Test.Hspec.Core.Extension.Config (Config)
+import qualified Test.Hspec.Core.Extension.Config as Config
+import qualified Test.Hspec.Core.Extension.Config.Type as Config
+import           Test.Hspec.Core.Extension.Item (Item)
+import           Test.Hspec.Core.Extension.Tree (SpecTree)
+
+newtype Option = Option { unOption :: Declarative.Option Config }
```

Comment:
You have several well organized sub extension modules like `Item`, `Spec`, `Tree`, and `Config`. The option types and functions would be better organized in their own module and then just have this a reexport module for all the sub export modules and the required types/functions.

Reply:
I haven't looked into the options yet, I try to get to it sometime soon.